domain/src/test/java/org/oppia/domain/oppialogger/logstorage/LogReportStorageHelperTest.kt

```diff
@@ -0,0 +1,208 @@
+    Dispatchers.resetMain()
+    testThread.close()
+  }
+
+  @InternalCoroutinesApi
+  @Test
+  @ExperimentalCoroutinesApi
+  fun testLogStorageHelper_addEventLog_addsLogInStore() {
+    logReportStorageHelper.addEventLog(eventLog)
+    val eventLogs = logReportStorageHelper.getEventLogs()
+
+    eventLogs.observeForever(mockOppiaEventLogsObserver)
+    testCoroutineDispatchers.advanceUntilIdle()
+
+    verify(mockOppiaEventLogsObserver, atLeastOnce())
+      .onChanged(oppiaEventLogsResultCaptor.capture())
+    assertThat(
+      oppiaEventLogsResultCaptor.value.getOrThrow()
+        .getEventLog(0)
+    ).isEqualTo(eventLog)
+  }
+
+  @InternalCoroutinesApi
+  @Test
+  @ExperimentalCoroutinesApi
+  fun testLogStorageHelper_addExceptionLog_addsLogInStore() {
+    logReportStorageHelper.addExceptionLog(exceptionLog)
+    val exceptionLogs = logReportStorageHelper.getExceptionLogs()
+
+    exceptionLogs.observeForever(mockOppiaExceptionLogsObserver)
+    testCoroutineDispatchers.advanceUntilIdle()
+
+    verify(mockOppiaExceptionLogsObserver, atLeastOnce())
+      .onChanged(oppiaExceptionLogsResultCaptor.capture())
+    assertThat(
+      oppiaExceptionLogsResultCaptor.value.getOrThrow()
+        .getExceptionLog(0)
+    ).isEqualTo(exceptionLog)
+  }
+
+  private fun setUpTestApplicationComponent() {
+    DaggerLogReportStorageHelperTest_TestApplicationComponent.builder()
+      .setApplication(ApplicationProvider.getApplicationContext())
+      .build()
+      .inject(this)
+  }
+
+  @Qualifier
+  annotation class TestDispatcher
+
+  // TODO(#89): Move this to a common test application component.
+  @Module
+  class TestModule {
+    @Provides
+    @Singleton
+    fun provideContext(application: Application): Context {
+      return application
+    }
+
+    @ExperimentalCoroutinesApi
+    @Singleton
+    @Provides
+    @TestDispatcher
+    fun provideTestDispatcher(): CoroutineDispatcher {
+      return TestCoroutineDispatcher()
+    }
+
+    // TODO(#59): Either isolate these to their own shared test module, or use the real logging
+    // module in tests to avoid needing to specify these settings for tests.
+    @EnableConsoleLog
+    @Provides
+    fun provideEnableConsoleLog(): Boolean = true
+
+    @EnableFileLog
+    @Provides
+    fun provideEnableFileLog(): Boolean = false
+
+    @GlobalLogLevel
+    @Provides
+    fun provideGlobalLogLevel(): LogLevel = LogLevel.VERBOSE
+  }
+
+  // TODO(#89): Move this to a common test application component.
+  @Singleton
+  @Component(
+    modules = [
+      TestDispatcherModule::class,
+      TestModule::class,
+      TestLogReportingModule::class,
+      LogStorageModule::class
+    ]
+  )
+  interface TestApplicationComponent {
+    @Component.Builder
+    interface Builder {
+      @BindsInstance
+      fun setApplication(application: Application): Builder
+
+      fun build(): TestApplicationComponent
+    }
+
+    fun inject(logReportStorageHelperTest: LogReportStorageHelperTest)
+  }
+}
```

Comment:
Please add EOF newlines in all new files.

Reply:
File removed.